testgen/generators.go

```diff
@@ -721,6 +956,80 @@ var EthEstimateGas = MethodTests{
 				return nil
 			},
 		},
+		{
+			Name:  "estimate-eip7702-single-auth",
+			About: "Estimates gas for a transaction with a single authorization in the EIP-7702 authorization list",
+			Run: func(ctx context.Context, t *T) error {
+				caller := t.chain.txinfo.EIP7702.Account
+				msg := map[string]any{
+					"from":  caller,
+					"to":    common.Address{0x01},
+					"value": hexutil.Uint64(10),
+					"authorizationList": []map[string]any{
+						{
+							"nonce":   hexutil.Uint64(1),
+							"chainId": hexutil.Uint64(t.chain.Config().ChainID.Uint64()),
+							"address": t.chain.txinfo.EIP7702.ProxyAddr,
+							"r":       (*hexutil.U256)(new(uint256.Int).SetUint64(87890)),
+							"s":       (*hexutil.U256)(new(uint256.Int).SetUint64(87890)),
+							"yParity": hexutil.Uint64(56),
+						},
+					},
+				}
+				var got hexutil.Uint64
+				err := t.eth.Client().CallContext(ctx, &got, "eth_estimateGas", msg, "latest")
+				if err != nil {
+					return err
+				}
+				//21000 + 25000(one address in auth list)
+				want := float64(46000)
+				if float64(got) > want*(1+estimateGasErrorRatio) {
```

Comment:
Please remove the check for a specific return value here. The important bit for this test is just ensuring that the `authorizationList` parameter is supported by the method `eth_estimateGas`. The return value is client-specific and we can't fully verify it.